qutebrowser/browser/webkit/network/proxy.py

```diff
@@ -23,18 +23,36 @@
 from PyQt5.QtNetwork import QNetworkProxy, QNetworkProxyFactory
 
 from qutebrowser.config import config, configtypes
+from qutebrowser.utils import objreg
 from qutebrowser.browser.webkit.network import pac
 
 
 def init():
     """Set the application wide proxy factory."""
-    QNetworkProxyFactory.setApplicationProxyFactory(ProxyFactory())
+    proxy_factory = ProxyFactory()
+    objreg.register('proxy-factory', proxy_factory)
```

Comment:
Well, I cheated when I said "no global flags" -- there _is_ now a global proxy factory accessible via `objreg` -- but it was global anyway.